editor/utils/auth.ts

```diff
@@ -59,33 +59,32 @@ export const formatRoleName = (role: string): string => {
 };
 
 /**
- * Hook which retrieves the user's access token and redirects to the
- * login page if the user's refresh token has expired
+ * Hook that returns a memoized function to get the Auth0 access token.
+ * Returns undefined if user is not authenticated or if token retrieval fails.
+ *
+ * @example
+ * const getToken = useGetToken();
+ * const fetchWithToken = async () => {
+ *  const token = await getToken();
+ *  if (token) {
+ *     fetch('/api/data', { headers: { Authorization: `Bearer ${token}` } });
+ *   }
+ * };
  */
-export const useToken = (): string | null => {
-  const pathName = usePathname();
-  const { getAccessTokenSilently, loginWithRedirect, isAuthenticated } =
-    useAuth0();
-  const [token, setToken] = useState<string | null>(null);
-  useEffect(() => {
-    const getToken = async () => {
-      if (!isAuthenticated) {
-        return;
-      }
-      try {
-        const accessToken = await getAccessTokenSilently();
-        setToken(accessToken);
-      } catch (err) {
-        console.error("Error getting access token:", err);
-        await loginWithRedirect({
-          appState: {
-            returnTo: BASE_PATH + pathName,
-          },
-        });
-      }
-    };
-    getToken();
-  }, [getAccessTokenSilently, isAuthenticated, loginWithRedirect, pathName]);
```

Comment:
as it turns out, neither `getAccessTokenSilently()` nor `loginWithRedirect()` are memoized and this was executing on every render 😖